state/executor.go

```diff
@@ -99,7 +99,14 @@ func (e *Executor) ProcessBlock(
 	}
 
 	txn.block = block
+
 	for _, t := range block.Transactions {
+		if t.ExceedsBlockGasLimit(block.Header.GasLimit) {
```

Comment:
This is very dangerous

You are omitting a state write for transactions that you have a receipt for (you're writing the receipt, but not writing the transaction itself to the transactions trie)

Please note that Ethereum writes failed transactions to state, along with their receipts

cc @lazartravica @Kourin1996

Reply:
I am not really sure what this means, so if we can sync on this

Reply:
You're writing a failed receipt to the receipts trie, but not the transaction to the transaction trie.

You can never look up these failed transactions, even though you can look up their receipts - which is not the case with Ethereum, where you can look up both receipts and transactions (failed)

Reply:
@lazartravica If you can, please check this. My explanation is that we do not change how we add receipt and transactions in block and state. This PR will add failed tx and receipt to tx[] receipt[] before we build block. Everything what happens after will work as it worked before. We would create block with tx[] and receipt[], Then we are going to insert it with insertBlock()...

Reply:
My mistake 🙏 

I've falsely assumed this is the final commit step, when in fact you write both failed and successful transactions when writing the block's body:

https://github.com/0xPolygon/polygon-edge/blob/d0dc886348bd134e35f03a2fe20930cae08c64e4/blockchain/blockchain.go#L637-L655